sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -165,6 +173,35 @@
  * <br>
  * 3. Start from <em>latest</em> offset by default;
  *
+ * <h3>Use Avro schema with Confluent Schema Registry</h3>
+ *
+ * <p>If the message keys or/and values in a topic were serialised by Avro schema stored in
+ * Confluent Schema Registry, KafkaIO source can fetch this schema from specified Schema Registry
+ * URL and use it for deserialization further. As an output, it will return a {@link PCollection} of
+ * {@link KafkaRecord}s where key or/and value will be typed as {@link
+ * org.apache.avro.generic.GenericRecord}. In this case, users don't need to specify key or/and
+ * value deserializers and coders since they will be set to {@link KafkaAvroDeserializer} and {@link
+ * AvroCoder} by default accordingly.
+ *
+ * <p>For example, below topic values are serialized with Avro schema stored in Schema Registry,
+ * keys are typed as {@link Long}:
+ *
+ * <pre>{@code
+ * // Confluent Schema Registry URL
+ * String schemaRegistryUrl = "http://localhost:8081";
```

Comment:
I would better let the documentation with the 3-arg method, prefer simpler use, we can easily evolve towards the more generic case if there is a need which so far seems not to be the case.